xonsh/xonfig.py

```diff
@@ -623,6 +637,53 @@ def _web(args):
     subprocess.run([sys.executable, "-m", "xonsh.webconfig"] + args.orig_args[1:])
 
 
+def _kernel(args):
+    """Make xonsh available as a Jupyter kernel."""
+    try:
+        from jupyter_client.kernelspec import KernelSpecManager
+    except ImportError:
+        raise ImportError("Jupyter not found in current Python environment")
```

Comment:
Probably should do a raise-from here: https://www.python.org/dev/peps/pep-3134/#id37